src/legacy/core_plugins/console/public/np_ready/lib/utils/utils.ts

```diff
@@ -87,8 +87,8 @@ export function expandLiteralStrings(data: string) {
       const colonAndAnySpacing = string.slice(0, firstDoubleQuoteIdx);
       const rawStringifiedValue = string.slice(firstDoubleQuoteIdx, string.length);
       const jsonValue = JSON.parse(rawStringifiedValue)
-        .replace('^s*\n', '')
-        .replace('\ns*^', '');
+        .replace('^\s*\n', '') // prettier-ignore
+        .replace('\n\s*$', ''); // prettier-ignore
```

Comment:
I think Prettier is correct here - these patterns be surrounded in forward-slashes.

```javascript
      const jsonValue = JSON.parse(rawStringifiedValue)
        .replace(/^\s*\n/, '')
        .replace(/\n\s*$/, '');
```

I see what you mean about the difference with `trim()` - it would remove all whitespace at the start or end of string, whereas these patterns are a little more subtle. Maybe it would be worth a comment, in case someone comes along later and replaces it with `trim()`. 😀

Reply:
(Although I still query the usefulness of this vs. a simple `trim()`)

Reply:
Come to think of it, this replace behaviour was part of legacy and was carried over from a previous alteration. I'm not convinced we want replace behaviour here at all. The value being altered was inside of `"""` which means string literal in Console. Newlines and whitespace should be preserved. I don't think those replace functions where actually _doing_ anything when the behaviour was considered "correct":

```javascript
"\n    SELECT * FROM \"TABLE\"\n    "
        .replace('^\s*\n', '')
        .replace('\n\s*$', '')

===

"\n    SELECT * FROM \"TABLE\"\n    "

// true
```

Automated and manual testing to confirm this. Would you mind taking a look too @pugnascotia ?